test/goalc/source_templates/with_game/test-vector-abs.gc

```diff
@@ -0,0 +1,23 @@
+(defun test-vector-abs ()
+  (let ((vector-0 (new 'stack 'vector))
+        (vector-1 (new 'stack 'vector)))
+		; TODO - a lib function get generate a vector would be useful
+		; TODO - it's really annoying getting a compiler error for passing `1` instead of `1.0`, quality of life improvement perhaps?
```

Comment:
It was somewhat intentional that floats/ints don't convert automatically in `set!` to help catch decompiler bugs. But it may be worth reconsidering, or at least having an option to disable this for code written by humans.